aggregator/aggregator.go

```diff
@@ -111,45 +111,41 @@ var _ core.Metricable = (*Aggregator)(nil)
 // TODO: Remove this context once OperatorPubkeysServiceInMemory's API is
 // changed and we can gracefully exit otherwise
 func NewAggregator(ctx context.Context, config *config.Config, logger logging.Logger) (*Aggregator, error) {
-	chainioConfig := eigenclients.BuildAllConfig{
-		EthHttpUrl:                 config.EthHttpRpcUrl,
-		EthWsUrl:                   config.EthWsRpcUrl,
-		RegistryCoordinatorAddr:    config.SFFLRegistryCoordinatorAddr.String(),
-		OperatorStateRetrieverAddr: config.OperatorStateRetrieverAddr.String(),
-		AvsName:                    avsName,
-		PromMetricsIpPortAddress:   config.MetricsIpPortAddress,
-	}
-	clients, err := eigenclients.BuildAll(chainioConfig, config.EcdsaPrivateKey, logger)
-	if err != nil {
-		logger.Error("Cannot create sdk clients", "err", err)
-		return nil, err
+	withOptionalMetrics := func(url string) safeclient.SafeEthClientOption {
+		if !config.EnableMetrics {
+			return func(*safeclient.SafeEthClient) {}
+		}
+
+		registry := prometheus.NewRegistry()
+		rpcCallsCollector := rpccalls.NewCollector(AggregatorNamespace+url, registry)
+		return safeclient.WithInstrumentedCreateClient(rpcCallsCollector)
 	}
-	registry := clients.PrometheusRegistry
 
-	ethHttpClient, err := core.CreateEthClientWithCollector(
-		AggregatorNamespace,
-		config.EthHttpRpcUrl,
-		config.EnableMetrics,
-		registry,
-		logger,
-	)
+	ethHttpClient, err := safeclient.NewSafeEthClient(config.EthHttpRpcUrl, logger, withOptionalMetrics(config.EthHttpRpcUrl))
 	if err != nil {
 		logger.Error("Cannot create http ethclient", "err", err)
 		return nil, err
 	}
 
-	ethWsClient, err := core.CreateEthClientWithCollector(
-		AggregatorNamespace,
-		config.EthWsRpcUrl,
-		config.EnableMetrics,
-		registry,
-		logger,
-	)
+	ethWsClient, err := safeclient.NewSafeEthClient(config.EthWsRpcUrl, logger, withOptionalMetrics(config.EthWsRpcUrl))
 	if err != nil {
 		logger.Error("Cannot create ws ethclient", "err", err)
 		return nil, err
 	}
 
+	clients, err := chainio.BuildAll(
```

Comment:
Within `chainio.BuildAll` you create `signerV2` & `txMgt` which are already created lower.

In this [PR](https://github.com/Layr-Labs/eigensdk-go/pull/117/files) I made `BuildAvsClient` & `BuildElClients` available for use. I would recommend using them instead of `BuildAll` and utilize `signerV2` & `txMgr` that are already created within this constructor

Reply:
Agree. This was a source of confusion when working in the PR. I'll refactor the code to not use `BuildAll` but rather what is exactly needed on each case avoiding duplicated work

Reply:
Integrated